### PR TITLE
Add check-release script

### PR DESCRIPTION
**Pull request by gerrod3**
[noissue]

### CODE REVIEW

File: templates/github/.ci/scripts/check_release.py.j2 [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    with TemporaryDirectory() as d:
+        # Clone from upstream to ensure we have updated branches & main
+        repo = Repo.clone_from(UPSTREAM_REMOTE, d, filter="blob:none")
+        heads = [h.split("/")[-1] for h in repo.git.ls_remote("--heads").split("\n")]
+        branches = [h for h in heads if re.search(RELEASE_BRANCH_REGEX, h)]
+        branches.sort(key=lambda ver: Version(ver))
+
+        versions = opts.branches
+        check_y_release = False
+        if versions == "supported":
```
Comment (mdellweg) [inline templates/github/.ci/scripts/check_release.py.j2:41]: This is a bit confusing. I would like to see the names "branch" and "version" be used consistently throughout the script.
Maybe the parameter should have been named versions? Or `branches` should have been `available_branches`?

File: templates/github/.ci/scripts/check_release.py.j2 [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            with open(f"{d}/template_config.yml", mode="r") as f:
+                tc = yaml.safe_load(f)
+                versions = tc["ci_update_branches"]
+            check_y_release = True
+        else:
+            splitted = versions.split(",")
+            versions = [v for v in splitted if re.match(RELEASE_BRANCH_REGEX, v) and v in branches]
```
Comment (mdellweg) [inline templates/github/.ci/scripts/check_release.py.j2:48]: `branches` was already regex filtered at this point.
Should we not fail on invalid input here?

File: templates/github/.ci/scripts/check_release.py.j2 [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            if len(versions) != len(splitted):
+                print("Supplied branches contains non-existent branches!")
+                exit(1)
+
+        print(
+            f"Checking for releases on branches: {versions}",
+            f"and {DEFAULT_BRANCH}" if check_y_release else "",
```
Comment (mdellweg) [inline templates/github/.ci/scripts/check_release.py.j2:55]: Can one check for the y-release without all the z-releases?

Is it even worth adding all this logic? We could just be opinionated and just always check all release branches and the default branch.

**Comment by gerrod3** [comment]
@mdellweg I refactored the script to address each of your concerns. Have another look.

**Review by mdellweg — APPROVED** [review]

**Review by ggainey — APPROVED** [review]
This looks good - but I will confess, I'll have to see it "in action" to be certain :)

**Pull request merged**